sklearn/ensemble/tests/test_weight_boosting.py

```diff
@@ -549,6 +543,26 @@ def test_adaboostregressor_sample_weight():
     assert score_no_outlier == pytest.approx(score_with_weight)
 
 
+@pytest.mark.parametrize(
+    "params, err_type, err_msg",
+    [
+        ({"n_estimators": -1}, ValueError, "n_estimators == 0, must be >= 1"),
+        ({"n_estimators": 0}, ValueError, "n_estimators == 0, must be >= 1"),
```

Comment:
Can you add a case where someone tries to a floating value? It should trigger a `TypeError` in this case.

Reply:
Definitely, I will do so now